source/documentation/tools/package-management.md

```diff
@@ -64,6 +121,27 @@ To activate Python integration within renv, type
 renv::use_python()
 ```
 
+### Common pitfalls with renv
+
+
+|     Situation    |     Why/What happens?    |
+|---|---|
+|     Packages   disappearing    |     You aren’t using renv!    |
+|     Forgetting to `renv::snapshot()`    |     This won’t affect you running your   code, but anyone picking it up later will be out of sync. You can use `renv::status()` to check if packages and renv.lock match        |
+|     Switching branches    |     If different package requirements in   branches then must remember to `renv::restore()`   when switching between them – otherwise library reflects the previous branch    |
+|     Initialising renv   outside a project    |     renv will ask you not to do this – do not   use `force   = TRUE`!    |
+|     Stuck on old CRAN/MRAN    |     Packages (or versions) you know exist   won’t appear using install functions. Run `options(repos   = "cran.rstudio.com")`    |
+
+### renv tips and tricks
+
+
+|     Situation    |     Solution    |
+|---|---|
+|     Got into a total mess?    |     Start again! Run `renv::deactivate()` and then delete the renv.lock file and the renv/ folder   |
+|     Add a package from github    |   Use `renv::install("username/packagename")` or for a private package `renv::install("git@github.com:username/packagename.git")` |
+|     Upgrade all packages to latest    |   Run `renv::update()` or `renv::update("packagename")` for specific package. Always check that upgrading packages does not break your code before pushing to github for other users. |
+|     Update renv itself    |     `renv::upgrade()`. Useful if renv gains new functionality that you want to use.    |
+
```

Comment:
```suggestion
|     Situation    |     Solution    |
|---|---|
|     Got into a total mess?    |     Start again! Run `renv::deactivate()` and then delete the renv.lock file and the renv/ folder   |
|     Add a package from github    |   Use `renv::install("username/packagename")` or for a private package `renv::install("git@github.com:username/packagename.git")` |
|     Upgrade all packages to latest    |   Run `renv::update()` or `renv::update("packagename")` for specific package. Always check that upgrading packages does not break your code before pushing to github for other users. |
|     Update renv itself    |     `renv::upgrade()`. Useful if renv gains new functionality that you want to use.    |
|     `Error in file(filename, "r", encoding = encoding) : 
  cannot open the connection`    |     You've accidentally installed `renv` in your home directory 🏠 ! Delete [all of the files created by `renv`](https://rstudio.github.io/renv/articles/renv.html#infrastructure) from your home directory and retry.    |
```

Reply:
@pjrh-moj added as a suggestion, not a commit. Feel free to adjust it however you see fit.